public/app/features/dashboard-scene/scene/DashboardScene.tsx

```diff
@@ -739,6 +737,7 @@ export class DashboardScene extends SceneObjectBase<DashboardSceneState> {
     sceneGridLayout.setState({
       children: [newGridItem, ...sceneGridLayout.state.children],
     });
+    this.setState({ body: sceneGridLayout });
```

Comment:
I assume this is to cause a state change and re-render of the DashboardSceneRenderer (since you removed the isEmpty state). 

There are many places more we would have to update as well I think (removeRow, removePanel, addPanel etc) ?